### PR TITLE
feat(`monitoringp`): implement `BeginBlock` and report signatures

### PR DESCRIPTION
**Pull request by lubtd**
Implement `BeginBlock` for the module where the validator signatures are reported and IBC packet is eventually transmitted

For testing: follow Monitoring Incentivized Testnets guide ;)

### CODE REVIEW

File: x/monitoringp/module.go [hunk 1]
```diff
@@ -168,7 +166,14 @@ func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.Raw
...
 
 // BeginBlock executes all ABCI BeginBlock logic respective to the capability module.
-func (am AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}
+func (am AppModule) BeginBlock(ctx sdk.Context, bb abci.RequestBeginBlock) {
+	// reports signatures for the block
+	am.keeper.ReportBlockSignatures(ctx, bb.LastCommitInfo, bb.Header.Height)
+
+	// check and transmit signatures
+	// TODO(): investigate proper error handling
+	_ = am.keeper.TransmitSignatures(ctx, bb.Header.Height)
```
Comment (Pantani) [inline x/monitoringp/module.go:175]:
```suggestion
	err := am.keeper.TransmitSignatures(ctx, bb.Header.Height)
	if err != nil {
		ctx.Logger().Error(fmt.Sprintf("error to transmit the validator signatures %s", err.Error()))
	}
```

File: x/monitoringp/keeper/begin_block.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	// check condition to transmit packet
+	// IBC connection to consumer must be established
+	// last block height must be reached
+	// monitoring info must exist
+	// signatures must not yet be transmitted
+	cid, cidFound := k.GetConnectionChannelID(ctx)
+	mi, miFound := k.GetMonitoringInfo(ctx)
+	if blockHeight < k.LastBlockHeight(ctx) || !cidFound || !miFound || mi.Transmitted {
+		return nil
+	}
```
Comment (Pantani) [inline x/monitoringp/keeper/begin_block.go:62]: only a suggestion, I think this way is more precise and avoid accessing the store if the  block height is not the right one

```suggestion
	if blockHeight < k.LastBlockHeight(ctx) {
		return nil
	}
	cid, cidFound := k.GetConnectionChannelID(ctx)
	if !cidFound {
		return nil
	}
	mi, miFound := k.GetMonitoringInfo(ctx)
	if !miFound || mi.Transmitted {
		return nil
	}
```

File: x/monitoringp/keeper/begin_block.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	valSetSize := int64(len(lastCommit.Votes))
+	for _, vote := range lastCommit.Votes {
+		if vote.SignedLastBlock {
+			// TODO(490): implement correct address format
+			monitoringInfo.SignatureCounts.AddSignature(vote.Validator.Address, valSetSize)
+		}
+	}
+
+	// increment block count and save the monitoring info
+	monitoringInfo.SignatureCounts.BlockCount++
```
Comment (Pantani) [inline x/monitoringp/keeper/begin_block.go:46]: Maybe we should add some logs here?

File: x/monitoringc/client/cli/query_monitoring_history.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/spf13/cast"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/spn/x/monitoringc/types"
+)
+
+func CmdListMonitoringHistory() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list-monitoring-history",
+		Short: "list all monitoringHistory",
```
[the remark this answers is not in the document]
Comment (lubtd) [reply x/monitoringc/client/cli/query_monitoring_history.go:16]: This is potentially a query not necessary https://github.com/tendermint/spn/issues/463

File: x/monitoringp/keeper/begin_block.go (hunk 2 above)
Comment (lubtd) [reply x/monitoringp/keeper/begin_block.go:62]: makes sense actually

File: x/monitoringp/module.go (hunk 1 above)
Comment (lubtd) [reply x/monitoringp/module.go:175]: That's a good suggestion
Also, FYI we have created a task to investigate the best way to handle error in begin block

File: x/monitoringp/keeper/begin_block.go (hunk 3 above)
Comment (lubtd) [reply x/monitoringp/keeper/begin_block.go:46]: We could although IMO log output from each block is already a lot verbose with Cosmso-SDK and the current state of the monitoring info can be queried with `q monitoring-provider show-monitoring-info`

File: x/monitoringc/client/cli/query.go [hunk 5]
```diff
@@ -34,6 +34,10 @@ func GetQueryCmd(queryRoute string) *cobra.Command {
 		CmdShowLaunchIDFromChannelID(),
 		CmdQueryParams(),
 	)
+	cmd.AddCommand(
```
Comment (Pantani) [inline x/monitoringc/client/cli/query.go:37]:
```suggestion
```

**Review by Pantani — DISMISSED** [review]

**Pull request merged**